### PR TITLE
[CS-2929]: Fix dust amount and migrate unclaimedRevenue

### PR DESCRIPTION
**Pull request by danibonilha**
### Description
This PR fixes the dust amount and moves the unclaimedRevenue from expandedState, to the new Sheet component. Abstracts the `RouteType` type to be shared across all screens. I wanted to test more on Android, but unable to build right now since Jcenter is out, but the code is good be reviewed already.

- [x] Completes #(CS-2929)

### Screenshots

Simulator Screen Recording - iPhone 11 - 2022-01-12 at 17 27 05

### CODE REVIEW

File: cardstack/src/components/TransactionList/TransactionListLoading.tsx [hunk 1]
```diff
@@ -15,8 +15,14 @@ const TransactionsSection = ({
...
 );
 
-export const TransactionListLoading = ({ light }: { light?: boolean }) => (
-  <Container padding={5} width="100%">
+export const TransactionListLoading = ({
+  light,
+  removePadding,
+}: {
+  removePadding?: boolean;
+  light?: boolean;
```
Comment (danibonilha) [inline cardstack/src/components/TransactionList/TransactionListLoading.tsx:23]: This is temporary, we can remove it when we migrate all the conditional sectionLists

File: src/components/expanded-state/UnclaimedRevenueExpandedState.tsx [hunk 2]
```diff
@@ -51,11 +52,7 @@ export default function UnclaimedRevenueExpandedState({
               <TokenItem key={token.tokenAddress} token={token} />
             ))}
           </Container>
-          <Button
-            marginTop={8}
-            onPress={isDust ? () => {} : onClaimAll}
-            variant={isDust ? 'disabled' : undefined}
-          >
+          <Button disabled={!isDust} marginTop={8} onPress={onClaimAll}>
```
Comment (lukemelia) [inline src/components/expanded-state/UnclaimedRevenueExpandedState.tsx]: Something seems off. You should not be able to claim dust, but this logic says the button should be disabled if it is NOT dust.

File: cardstack/src/screens/sheets/UnclaimedRevenue/UnclaimedRevenueSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    [revenueBalances]
+  );
+
+  return (
+    <Container paddingHorizontal={5} paddingVertical={3}>
+      <Text size="medium">Claim available revenue</Text>
+      <Container flexDirection="column" marginTop={5}>
+        {renderTokens}
+      </Container>
+      <Button disabled={!isDust} marginTop={8} onPress={onClaimAll}>
```
Comment (lukemelia) [inline cardstack/src/screens/sheets/UnclaimedRevenue/UnclaimedRevenueSheet.tsx]: Same here. It should be disabled if there is nothing to claim or only dust to claim.

File: cardstack/src/screens/sheets/UnclaimedRevenue/UnclaimedRevenueSheet.tsx [hunk 4]
```diff
@@ -18,40 +18,25 @@ import { ClaimStatuses } from '@cardstack/utils';
 import { RouteType } from '@cardstack/navigation/types';
 
 interface Params {
-  asset: MerchantSafeType;
-  customFunction: () => void;
+  merchantSafe: MerchantSafeType;
+  onClaimAllPress: () => void;
```
Comment (lukemelia) [inline cardstack/src/screens/sheets/UnclaimedRevenue/UnclaimedRevenueSheet.tsx:22]: 👍  Much better name :)

**Review by lukemelia — CHANGES_REQUESTED** [review]

File: src/components/expanded-state/UnclaimedRevenueExpandedState.tsx (hunk 2 above)
Comment (danibonilha) [reply src/components/expanded-state/UnclaimedRevenueExpandedState.tsx]: my bad, I committed the wrong piece 🤦‍♀️

Comment (danibonilha) [reply src/components/expanded-state/UnclaimedRevenueExpandedState.tsx]: `isDust` should be `hasClaimableAmount`, bc on gnosis safe we already check for dustAmount, and if we have one, the native amount is set to zero, so in unclaimedRevenue we check if native amount is a valid value. 
`const hasClaimableAmount = !!nativeAmount`. 
The previous name was misleading as we are not checking for dust anymore.

**Review by lukemelia — APPROVED** [review]
🙌  LGTM!

**Pull request merged**